wire/internal/wire/analyze.go

```diff
@@ -388,36 +391,43 @@ func verifyAcyclic(providerMap *typeutil.Map, hasher typeutil.Hasher) []error {
 				continue
 			}
 			visited.Set(head, true)
-			switch x := providerMap.At(head).(type) {
-			case nil:
+			x := providerMap.At(head)
+			if x == nil {
 				// Leaf: input.
-			case *Value:
+				continue
+			}
+			pt, ok := x.(*ProvidedType)
```

Comment:
If you're going to use the `ok` value just to panic, it is more concise to just type assert without the `ok` value.

Reply:
Done.